File: algo/lc.136.go
package main

import "fmt"

func singleNumber(nums []int) int {
	var ret int = 0
	for _, v := range nums {
		ret ^= v
	}
	return ret
}

func main() {
	a := []int{1, 1, 2, 3, 3}
	b := []int{1, 2, 3, 4, 5, 5, 4, 3, 2}

	fmt.Println(singleNumber(a))
	fmt.Println(singleNumber(b))
}

File: algo/lc.136.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return reduce(lambda x,y: x^y, nums)

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        res = 0
        for x in nums:
            res = res ^ x
        return res
